src/OrchardCore.Modules/OrchardCore.Users/Views/UserButtons.cshtml

```diff
@@ -5,7 +5,14 @@
     var user = Model.User as User;
 }
 <a asp-action="Edit" asp-route-id="@user.Id" class="btn btn-primary btn-sm">@T["Edit"]</a>
-<a asp-action="Delete" asp-route-id="@user.Id" class="btn btn-danger btn-sm" itemprop="RemoveUrl UnsafeUrl">@T["Delete"]</a>
+@if (user.UserName == User.Identity.Name)
+{
+    <a class="btn btn-danger btn-sm disabled" itemprop="RemoveUrl UnsafeUrl">@T["Delete"]</a>
```

Comment:
Something like 
`<a class="btn btn-danger btn-sm @((user.UserName == User.Identity.Name) ? "disabled" : "")" itemprop="RemoveUrl UnsafeUrl">@T["Delete"]</a>` 
could avoid duplicating the line.

But it could be better to do it in the controller and return an error message if it occurs.

Same for the other changes.

Reply:
I already did something similar to what you did but I faced an issues with the TagHelper, anyhow I will try to have a look again